applications/rag/README.md

```diff
@@ -0,0 +1,137 @@
+# RAG-on-GKE Application
+
+This is in beta/a work in progress.
+
+This is a sample to deploy a RAG application on GKE. Retrieval Augmented Generation (RAG) is a popular approach for boosting the accuracy of LLM responses, particularly for domain specific or private data sets. The basic idea is to have a semantically searchable knowledge base (often using vector search), which is used to retrieve relevant snippets for a given prompt to provide additional context to the LLM. Augmenting the knowledge base with additional data is typically cheaper than fine tuning and is more scalable when incorporating current events and other rapidly changing data spaces.
+
+Architecture:
+1) A k8s service serving Hugging Face TGI inference using `mistral-7b`.
+2) Cloud SQL `pgvector` instance with vector embeddings generated from the input dataset.
+3) A k8s front end chat interface to interact with the inference server and the vector DB.
+4) Ray cluster runs job to populate vector DB.
+5) Jupyter notebook to read dataset and trigger ray job to populate the vector DB.
+
+## Installation
+
+Preinstall the following on your computer:
+* Kubectl
+* Terraform
+* Helm
+* Gcloud
+
+> **_NOTE:_** Terraform keeps state metadata in a local file called `terraform.tfstate`. Deleting the file may cause some resources to not be cleaned up correctly even if you delete the cluster. We suggest using `terraform destory` before reapplying/reinstalling.
+
+### Infra Setup
+
+#### GKE Cluster Setup
+1. This sample assumes a GKE cluster already exists. If not, please follow the instructions at https://github.com/GoogleCloudPlatform/ai-on-gke/infrastructure/README.md to create a GKE cluster.
+
+TODO (b/325443607): Min GKE cluster requirements for a successful installation.
+
+2. Ensure your environment is pointing at the created cluster by running `gcloud container clusters get-credentials <cluster-name>` --location=<region or zone>.
+
+3. The inference server requires L4 GPUs. Create an additional node pool:
+```
+gcloud container node-pools create g2-standard-24 --cluster <cluster-name> \
+  --accelerator type=nvidia-l4,count=2,gpu-driver-version=latest \
+  --machine-type g2-standard-24 \
+  --ephemeral-storage-local-ssd=count=2 \
+ --enable-image-streaming \
+ --num-nodes=1 --min-nodes=1 --max-nodes=2 \
+ --node-locations $REGION-a,$REGION-b --region $REGION
```

Comment:
Does this need to be regional ? Can I just have a single node instead? A lot of GPU resources are expensive, so it is better to create less resources.